src/bio.c

```diff
@@ -79,6 +79,8 @@ static unsigned long long bio_pending[BIO_NUM_OPS];
 struct bio_job {
     /* Job specific arguments.*/
     int fd; /* Fd for file based background jobs */
+    unsigned need_fsync:1; /* A flag to indicate that a fsync is required before
+                            * the file is closed. */
```

Comment:
maybe this struct should become a union?
this one bit still consumes 4 bytes, the fact is that even without it, we had 4 bytes of padding, but when we look at this, the `free_*` args and the `fd` / `fsync` args are never used together.

on the other hand the memory footprint of this struct is meaningless, so this cleanup is only worth it if it doesn't cause too much mess for existing lines.

Reply:
i try to make it become a union in the last commit, let's see if we need it